### PR TITLE
DAT-79160 select remove selection ability

### PR DESCRIPTION
**Pull request by mlevi1806**
## Description
remove selection ability by using `isSingleOptionRemovable` flag
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Are you adding a new package?
- [ ] yes
- [x] no

### CODE REVIEW

File: src/components/Select/SelectHeader.js [hunk 1]
```diff
@@ -68,6 +72,25 @@ const SelectHeader = (props) => {
...
   const headerLoading = !searchable && loading;
 
+  const icon = useMemo(() => {
+    if (headerLoading) {
+      return <StyledSelectSpinner />;
+    }
+
+    if (values.length > 0 && isSingleOptionRemovable && !inlineSearch) {
+      return <RemoveIcon onClick={onClearSelect} className="remove-icon" />;
+    }
```
Comment (Chikel) [inline src/components/Select/SelectHeader.js:82]: So if I want to select another option, I have to clear the selected one first? (only then will I see the Arrow icon)
Any chance they can "live" next to each other?

**Review by Chikel — CHANGES_REQUESTED** [review]

File: src/components/Select/SelectHeader.js (hunk 1 above)
Comment (mlevi1806) [reply src/components/Select/SelectHeader.js:82]: You don't need to clear the selection first, you can simply select another option.
As long as there is a selected option and the `isSingleOptionRemovable` is `true` , the remove icon is replacing the arrow icon.
showing both icons is quite confusing

Comment (Chikel) [reply src/components/Select/SelectHeader.js:82]: I wonder how will be the UX, if the users will understand they can click the value to open the menu.

**Review by Chikel — APPROVED** [review]

**Pull request merged**